### PR TITLE
Update React Structural Metadata Editor

### PR DESCRIPTION
**Pull request by jefferya**
Address #680, #697 

* updates React Structural Metadata Editor (SME) to the latest version
  * https://github.com/avalonmediasystem/react-structural-metadata-editor#9e9527eb4c5ae2f17a69441e61955743cb0d5117
* update Ruby connector to support updated (SME): 
  * unsaved change handling: cherry-pick files but leave yarn.lock: https://github.com/avalonmediasystem/avalon/commit/3d569fb2a3b4d5be2b1f040db9abacd53f8a8988
 * update hls.js
 * update Dockerfile to NodeJS v10/12 as SME doesn't support v8 specified in Dockerfile
 * update dependencies

### CODE REVIEW

File: app/views/media_objects/_structure.html.erb [hunk 1]
```diff
@@ -69,7 +69,7 @@ Unless required by applicable law or agreed to in writing, software distributed
               <input type="button" class="btn btn-primary btn-xs" onclick="$('#structure_<%= index %> .filedata').click();" value="Upload" />
               <div class="btn-group">
                 <% if section.structuralMetadata.valid? %>
-                  <%= react_component("ReactButtonContainer", {sectionIndex: index, masterFileID: section.id, baseURL: request.protocol+request.host_with_port, streamDuration: section.duration, audioStreamURL: stream_url}) %>
+                  <%= react_component("ReactButtonContainer", {sectionIndex: index, masterFileID: section.id, baseURL: request.protocol+request.host_with_port, streamDuration: section.duration, audioStreamURL: stream_url, initStructure: section.structuralMetadata.as_json}) %>
```
Comment (jefferya) [inline app/views/media_objects/_structure.html.erb:72]: fixes a breaking change in react-structural-metadata-editor js package

File: app/javascript/components/ReactButtonContainer.jsx [hunk 2]
```diff
@@ -15,12 +15,19 @@ class ReactButtonContainer extends Component {
         initStructure: this.props.initStructure,
         audioStreamURL: this.props.audioStreamURL,
         streamDuration: this.props.streamDuration
-      }
+      },
+      structureSaved: true
```
Comment (jefferya) [inline app/javascript/components/ReactButtonContainer.jsx:19]: Changes to this file fixe a breaking change in react-structural-metadata-editor js package

File: package.json [hunk 3]
```diff
@@ -8,23 +8,24 @@
     "react": "^16.8.6",
     "react-bootstrap": "^1.0.0-beta.8",
     "react-dom": "^16.8.6",
-    "react-structural-metadata-editor": "https://github.com/avalonmediasystem/react-structural-metadata-editor#avalon-r6.5",
+    "react-structural-metadata-editor": "https://github.com/avalonmediasystem/react-structural-metadata-editor",
```
Comment (jefferya) [inline package.json:11]: Remove version pin for the Structural Metadata Editor (SME). Upstream doesn't follow semver release tagging. yarn.lock pins to a specific commit.

File: docker/Dockerfile [hunk 4]
```diff
@@ -3,9 +3,9 @@ MAINTAINER  Phuong Dinh <pdinh@indiana.edu>
 
 RUN         curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - \
          && echo "deb http://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list \
-         && curl -sL http://deb.nodesource.com/setup_8.x | bash - \
+         && curl -sL http://deb.nodesource.com/setup_10.x | bash - \
          && echo "deb http://deb.debian.org/debian stretch-backports main" >> /etc/apt/sources.list \
-         && wget https://mediaarea.net/repo/deb/repo-mediaarea_1.0-6_all.deb && dpkg -i repo-mediaarea_1.0-6_all.deb
+         && wget https://mediaarea.net/repo/deb/repo-mediaarea_1.0-14_all.deb && dpkg -i repo-mediaarea_1.0-14_all.deb
```
Comment (jefferya) [inline docker/Dockerfile:8]: Update NodeJS to v10, same version used by production

**Review by murny — APPROVED** [review]
LGTM 👍

File: docker-compose.yml [hunk 5]
```diff
@@ -76,7 +76,7 @@ services:
       - "6379:6379"
 
   avalon:
-    image: avalonmediasystem/avalon:6.x-dev
+    # image: avalonmediasystem/avalon:6.x-dev
```
Comment (pgwillia) [inline docker-compose.yml:79]: Is this an intended related change?

Comment (jefferya) [reply docker-compose.yml:79]: Yes, upstream is no longer updating 6x Docker images and the tag locks NodeJS to v8 which is end-of-life and unsupported by some nodejs packages in this update.
